### ISSUE TITLE
NumericUpDown didn't update when lost focus, StringFormat  has no effect to the content string

### ISSUE THREAD

**Issue by ShannonZ** [MahApps/MahApps.Metro#3026]
# NumericUpDown didn't update

I‘ve wrote an simple solution to  demonstrate this issue [download code](https://drive.google.com/open?id=0B9QI8E5qFmuEYzhfcXRJb3V0SEU). 

Value of  NumericUpDown(NUD) was bonded to the field of Model class
-`public float Age { set; get; } `
And the StringFormat of NUD was set as
-`StringFormat="{}{0:N6} ms"`

## if one typed something like "9.87654321" and tab to another control, the content of NUD will still be "9.87654321", not the exact one "9.876543 ms"

The thing is quirk because the NUD used in the offical demo solution (right-bottom in the Text tab) present well. If you input "9.87654321" and then change the focus, the content of NUD will update as expected immediately.

### Environment

- MahApps.Metro __v1.5.0__
- Windows OS __win7 x64__
- Visual Studio __2015__
- .NET Framework __4.5__
- Reference  [SoftFluent.Windows](https://github.com/SoftFluent/SoftFluent.Windows)

**Comment by punker76** [comment]
@ShannonZ I can not reproduce this with the v1.5.0 of MahApps.

![numericupdown_3026](https://user-images.githubusercontent.com/658431/29603061-fa07b6d0-87e2-11e7-8474-fc25839c492b.gif)

**Comment by ShannonZ** [comment]
just type more digits to call up the Horizotal Scrollbar of NUD.

**Comment by ShannonZ** [comment]
> @ShannonZ I can not reproduce this with the v1.5.0 of MahApps.

I have done this experiment in another two computers with VS2015 installed, but the problem is still exist.
Is is possible to relate with cultureinfo?

**Comment by ShannonZ** [comment]
![1](https://user-images.githubusercontent.com/7865691/29654815-4b2437a6-88e1-11e7-9499-abc36511f4b4.png)

**Comment by punker76** [comment]
@ShannonZ where/what is the code of this?

**Comment by ShannonZ** [comment]
@punker76  The solution attached in the first floor has the same problem.
[Solution](https://drive.google.com/file/d/0B9QI8E5qFmuEYzhfcXRJb3V0SEU/view?usp=drive_web)

**Comment by punker76** [comment]
@ShannonZ In this sample you set `StringFormat="0"` if I set it to `StringFormat="{}{0:N6} ms"` it works for me™.

**Comment by ShannonZ** [comment]
@punker76  Have you input enough digits? Have you seen the horizontal scrollbar?

**Comment by ShannonZ** [comment]
[The new](https://drive.google.com/open?id=0B9QI8E5qFmuEUnJHQzIwNEdSdUk)

**Comment by ShannonZ** [comment]
How to update the content of NUD programmatically?

**Comment by punker76** [comment]
@ShannonZ to hide the scrollbars do this on your NumericUpDown
```
ScrollViewer.HorizontalScrollBarVisibility="Hidden"
ScrollViewer.VerticalScrollBarVisibility="Hidden"
```

**Comment by punker76** [comment]
@ShannonZ íf there is no other tabable control the tab doesn't work

**Comment by ShannonZ** [comment]
Can the problem be reproduce at your computer?
In my office, only if inputting excessive precision the content of NUD won't be update any more.
But the button(+/-) can work perfectly. Is it possible to call the Click callback of + programtocally? Succeeding with a calling of Click callback of -.

**Comment by punker76** [comment]
@ShannonZ I'm investigating what's wrong with this

**Comment by ShannonZ** [comment]
@punker76  Has it been fixed？

**Comment by ShannonZ** [comment]
@punker76  Is there any  temporary solution?
Sorry to disturb you so frequently but I myself cannot solve it.

**Comment by punker76** [comment]
@ShannonZ Should be fixed now with bd5b84f7ce74d0d36494cb9168059930fadf4441, available in 1.6.0 (and upcoming alpha)
